app/frontend/src/components/Stats.vue

```diff
@@ -1,79 +1,113 @@
+      <v-card-title>
+        Registered Service Clients
+        <v-spacer></v-spacer>
+        <v-text-field
+          v-model="search"
+          append-icon="mdi-magnify"
+          label="Search"
+          single-line
+          hide-details
+        ></v-text-field>
+      </v-card-title>
+
       <v-data-table
+        class="kc-table"
+        dense
         :headers="headers"
         :items="serviceClients"
-        :items-per-page="2"
-      ></v-data-table>
+        :items-per-page="5"
+        :search="search"
+        :loading="loading"
+        loading-text="Loading... Please wait"
+      >
+        <template v-slot:item.dev="{item}">
+          <v-icon v-if="item.dev === 1" color="green">check</v-icon>
+        </template>
+
+        <template v-slot:item.test="{item}">
+          <v-icon v-if="item.test === 1" color="green">check</v-icon>
+        </template>
+
+        <template v-slot:item.prod="{item}">
+          <v-icon v-if="item.prod === 1" color="green">check</v-icon>
+        </template>
+      </v-data-table>
     </template>
-  </v-container>
+  </v-card>
 </template>
 
 
 <script>
-import statsService from '@/services/statsService';
+import keycloakService from '@/services/keycloakService';
 
 export default {
   name: 'Stats',
   data() {
     return {
+      // vuetify data table
+      search: '',
       headers: [
-        { text: 'Service Client', align: 'start', value: 'name' },
-        { text: 'Created', value: 'created' },
+        { text: 'Name', align: 'start', value: 'name' },
         { text: 'DEV', value: 'dev' },
         { text: 'TEST', value: 'test' },
         { text: 'PROD', value: 'prod' }
       ],
-      serviceClients: this.getData()
-      /*
-      [
-        {
-          name: 'ABC_ONE',
-          created: 159,
-          dev: 1,
-          test: 0,
-          prod: 0
-        },
-        {
-          name: 'DEF_TWO',
-          created: 237,
-          dev: 1,
-          test: 1,
-          prod: 0
-        },
-        {
-          name: 'HKI_THREE',
-          created: 262,
-          dev: 1,
-          test: 0,
-          prod: 0
-        },
-        {
-          name: 'BED_FOUR',
-          created: 305,
-          dev: 1,
-          test: 1,
-          prod: 1
-        }
-      ]
-      */
+      serviceClients: [],
+      loading: true
     };
   },
-
+  watch: {
+    // hide data table progress bar when serviceClients have been returned from backend
+    serviceClients() {
+      this.loading = false;
+    }
+  },
   methods: {
     // get table data from frontend service layer
     getData() {
-      statsService
-        .getServiceClients()
-        .then(response => {
-          if (response) {
-            return response;
+      keycloakService.getServiceClients().then(response => {
+        if (response) {
+
+          // reformat serviceClients data into array of objects that will work for the data table
+          var output = [];
+          var tmpArray = [];
```

Comment:
Always use `const` if possible, or `let` if  the variable needs to be reassigned.
You likely don't need both output and tmpArray - just pick one of them and stick with it.

Reply:
good call